### ISSUE TITLE
cookbook rabbitmq doesn't enable UI and restart service

### ISSUE THREAD

**Issue by vijaykanthm28** [megamsys/chef-repo#83]
t on service[rabbitmq-server](delayed), as it's already been queued\n  \* execute[Enable rabbitmq-plugin for rabbitmq_management ] action run[2016-03-18T13:07:44+00:00] INFO: Processing execute[Enable rabbitmq-plugin for rabbitmq_management ] action run (megam_rabbitmq::default line 118)\n\n       

================================================================================   \x1b[31m
Error executing action `run` on resource 'execute[Enable rabbitmq-plugin for rabbitmq_management ]'   ================================================================================

```
   Mixlib::ShellOut::ShellCommandFailed   ------------------------------------   Expected process to exit with [0], but received '1'    ---- Begin output of rabbitmq-plugins enable rabbitmq_management ----    STDOUT:     STDERR: erlexec: HOME must be set    ---- End output of rabbitmq-plugins enable rabbitmq_management ----    Ran rabbitmq-plugins enable rabbitmq_management returned 1       Resource Declaration:   ---------------------   # In /var/lib/megam/gulp/chef-repo/cookbooks/megam_rabbitmq/recipes/default.rb        118: execute \"Enable rabbitmq-plugin for rabbitmq_management \" do    119:   command \"rabbitmq-plugins enable rabbitmq_management\"    120: end    121:         Compiled Resource:   ------------------   # Declared in /var/lib/megam/gulp/chef-repo/cookbooks/megam_rabbitmq/recipes/default.rb:118:in `from_file'        execute(\"Enable rabbitmq-plugin for rabbitmq_management \") do      action \"run\"      retries 0      retry_delay 2      default_guard_interpreter :execute      command \"rabbitmq-plugins enable rabbitmq_management\"      backup 5      returns 0      declared_type :execute      cookbook_name :megam_rabbitmq      recipe_name \"default\"    end    [2016-03-18T13:07:44+00:00] INFO: Running queued delayed notifications before re-raising exception\n[2016-03-18T13:07:44+00:00] INFO: template[/etc/rabbitmq/rabbitmq-env.conf] sending restart action to service[rabbitmq-server] (delayed)\n  * service[rabbitmq-server] action restart[2016-03-18T13:07:44+00:00] INFO: Processing service[rabbitmq-server] action restart (megam_rabbitmq::default line 178)\n\n       ================================================================================   \
```

x1b[31mError executing action `restart` on resource 'service[rabbitmq-server]'  
================================================================================       Mixlib::ShellOut::ShellCommandFailed   ------------------------------------   Expected process to exit with [0], but received '1'    ---- Begin output of stop rabbitmq-server && start rabbitmq-server ----    STDOUT:     STDERR: stop: Unknown job: rabbitmq-server    ---- End output of stop rabbitmq-server && start rabbitmq-server ----    Ran stop rabbitmq-server && start rabbitmq-server returned 1       Resource Declaration:   ---------------------   # In /var/lib/megam/gulp/chef-repo/cookbooks/megam_rabbitmq/recipes/default.rb        178:     service node['rabbitmq']['service_name'] do    179:       provider Chef::Provider::Service::Upstart    180:       action [:enable, :start]    181:       restart_command \"stop #{node['rabbitmq']['service_name']} && start #{node['rabbitmq']['service_name']}\"    182:       only_if { File.exist?('/etc/init.d/rabbitmq-server') }    183:     end    184:           Compiled Resource:   ------------------   # Declared in /var/lib/megam/gulp/chef-repo/cookbooks/megam_rabbitmq/recipes/default.rb:178:in `from_file'        service(\"rabbitmq-server\") do      provider Chef::Provider::Service::Upstart      action [:enable, :start]      supports {:restart=>false, :reload=>false, :status=>false}      retries 0      retry_delay 2      default_guard_interpreter :default      service_name \"rabbitmq-server\"      pattern \"rabbitmq-server\"      restart_command \"stop rabbitmq-server && start rabbitmq-server\"      declared_type :service      cookbook_name :megam_rabbitmq      recipe_name \"default\"      only_if { #code block }    end    \x1b[0m\nRunning handlers:\x1b[0m\n[2016-03-18T13:07:44+00:00] ERROR: Running exception handlers\nRunning handlers complete[2016-03-18T13:07:44+00:00] ERROR: Exception handlers complete\n[2016-03-18T13:07:44+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out\nChef Client failed. 12 resources updated in 465.454285083 seconds\x1b[0m\n[2016-03-18T13:07:44+00:00] ERROR: Chef::Exceptions::MultipleFailures\n[2016-03-18T13:07:45+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)\n  chefsolo run ended failed.\nexit status 1" 
time="2016-03-18T13:07:45Z" level=error msg="Unable to stateup box : exit status 1" 
time="2016-03-18T13:07:45Z" level=debug msg="nicknames.megambox.com_log:{18 Mar 16 13:07 UTC   chefsolo run ended failed. gulp subsides COM6943313236541040135}" 
time="2016-03-18T13:07:45Z" level=debug msg="nicknames.megambox.com_log:{18 Mar 16 13:07 UTC exit status 1 gulp subsides COM6943313236541040135}"

**Comment by indykish** [comment]
@vijaykanthm28  can we close this, as this is fixed by @vinomca-megam  ?

**Comment by vijaykanthm28** [comment]
Issue was fixed by the pull request #103
